src/components/tl/nccl/tl_nccl_coll.c

```diff
@@ -143,6 +144,11 @@ ucc_status_t ucc_tl_nccl_init_task(ucc_base_coll_args_t *coll_args,
         return UCC_ERR_NOT_SUPPORTED;
     }
 
+    status = ucc_tl_nccl_comm_init(nccl_team);
```

Comment:
maybe we can call this func only if (team->comm_state != TL_NCCL_COMM_STATE_READY)  to avoid unnecessary for every task init.

Reply:
added